src/bin/main.rs

```diff
@@ -564,6 +601,20 @@ impl GetOptsOptions {
         }
 
         options.config_path = matches.opt_str("config-path").map(PathBuf::from);
+        options.stdin_file_hint = matches.opt_str("stdin-file-hint").map(PathBuf::from);
+
+        // return early if there are issues with the file hint specified
+        if let Some(file_hint) = &options.stdin_file_hint {
+            if !file_hint.exists() {
```

Comment:
Is there a reason that the file must exist on the file system?
For our use case https://github.com/rust-lang/rustfmt/issues/5792 (files in git history) there could be egde cases where the file is not present on disk.

Reply:
To be honest, it's been a very long time since I've looked at this PR, and I don't remember why. I definitely didn't have your use case in mind when I initially wrote this PR.

It's hard to say when I'll revisit this, but when I do, I'll double check if there are any issues with providing non-existent files as the file hint.

Reply:
Another use case would be editing usaved files.